src/D2L.CodeStyle.Analysis/MutabilityInspector.cs

```diff
@@ -125,7 +125,7 @@ HashSet<ITypeSymbol> typeStack
 
 				{
 					foreach( ISymbol member in type.GetMembers() ) {
-						if( IsMemberMutableRecursive( member, typeStack ) ) {
+						if( !member.IsStatic && IsMemberMutableRecursive( member, typeStack ) ) {

```

Comment:
In the case of `Foo`, it is not just the `static` that makes it safe.  The `readonly` is also needed.

Consider your second case where `Default` is not `readonly`.  No longer safe.

What about 

```
public class Foo {
    public static readonly Bar Default = new Bar();
}
```

where `Bar` is mutable?